Yosemite/Yosemite/SnapshotsProvider/FetchResultSnapshotsProvider.swift

```diff
@@ -0,0 +1,156 @@
+import Storage
+import CoreData
+import Combine
+
+/// The mutable (`Storage` model) type that is used by `FetchResultSnapshotsProvider`.
+public typealias FetchResultSnapshotsProviderMutableType = NSManagedObject & ReadOnlyConvertible
+
+/// The type of the items emitted by `FetchResultSnapshot`.
+///
+/// This is only declared to _hide_ `NSManagedObjectID` from the UI.
+public typealias FetchResultSnapshotObjectID = NSManagedObjectID
```

Comment:
Maybe `StorageObjectID` or will this only be needed in the `FetchResultsSnapshop` context?

Reply:
It looks like we could use this:

https://github.com/woocommerce/woocommerce-ios/blob/de9394f0a67236e199322cd309743ae3a2e4c47d/Storage/Storage/Protocols/Object.swift#L8-L10

But it doesn't seem to be a simple refactor. I'll take a look at migrating to that on another PR. 

Thanks for the nudge!